File: core/Base7/Base7.java
package com.dbc;

public class Base7 {
    public String convertToBase7(int num) {
        if (num == 0) {
            return "0";
        }

        StringBuffer res = new StringBuffer();
        int temp = Math.abs(num);
        while (temp != 0) {
            res.append(temp % 7);
            temp /= 7;
        }

        return num >= 0 ? res.reverse().toString() : res.append("-").reverse().toString();
    }
}

File: core/Base7/Base7.py
def convertToBase7(num: int) -> str:
    if num == 0:
        return "0"

    res = ""
    temp = abs(num)
    while temp != 0:
        res = str(temp % 7) + res

        temp = temp // 7

    return res if num >= 0 else f"-{res}"


if __name__ == '__main__':
    print(convertToBase7(0))
